frontend/src/api/Search.js

```diff
@@ -1,137 +1,151 @@
 import axios from 'axios';
-import { timeout } from 'd3';
-
-// Given a rapidwords domain or index, returns a json object defining the hyponyms and hypernym of the domain
-// Example response:
-// {
-//   "index": "6.3.1",
-//   "domain": "Domesticated animal",
-//   "hypernyms": [
-//       {"index": "6.3", "domain": "Animal husbandry"}
-//   ],
-//   "hyponyms": [
-//       {"index": "6.3.1.1", "domain": "Cattle"},
-//       {"index": "6.3.1.2", "domain": "Sheep"},
-//       {"index": "6.3.1.4", "domain": "Goat"},
-//       {"index": "6.3.1.5", "domain": "Pig"},
-//       {"index": "6.3.1.6", "domain": "Dog"},
-//       {"index": "6.3.1.7", "domain": "Cat"},
-//       {"index": "6.3.1.8", "domain": "Beast of burden"}
-//   ]
-// }
-export function searchRWRelations(domain){
-  if (!domain) return;
-  
-  return axios.get('https://api.itwewina.altlab.dev/api/rapidwords/', {
-    params: {
-      q: domain
+import rapidwordsData from '../data/rapidwords.json'; // Already parsed JSON, no need for JSON.parse
+
+console.log(rapidwordsData);
+
+// Given a rapidwords index, returns a JSON object defining the domain
+export async function searchRWRelations(index) {
+  if (!index) return;
+
+  try {
+    const response = await axios.get('https://itwewina.altlab.dev/api/rapidwords-index/', {
+      params: { rw_index: index }
+    });
+
+    const results = response.data.results;
+    if (!results || results.length === 0) return {};
+
+    // Extract domain name based on the index position
+    const rwDomains = results[0]?.lemma_wordform?.rw_domains || [];
+    const rwIndices = results[0]?.lemma_wordform?.rw_indices || [];
+
+    let domain = "Unknown Domain";
+    const indexPosition = rwIndices.indexOf(index);
```

Comment:
Although this approach is O(n), it's not going to be a problem in practice because rwIndices tends to be small